swap/src/protocol/alice/event_loop.rs

```diff
@@ -148,25 +147,25 @@ where
                 swarm_event = self.swarm.next_event() => {
                     match swarm_event {
                         SwarmEvent::Behaviour(OutEvent::SpotPriceRequested { request, channel, peer }) => {
-                            if self.resume_only {
-                                tracing::warn!(%peer, "Ignoring spot price request from {} because ASB started in resume-only mode", peer);
-
-                                match self.swarm.behaviour_mut().spot_price.send_response(channel, spot_price::Response { xmr: None, error: Some(spot_price::Error::MaintenanceMode) }) {
-                                    Ok(_) => {},
-                                    Err(_) => {
-                                        tracing::debug!(%peer, "Failed to respond with error to spot price request");
-                                        continue;
-                                    }
-                                }
-
-                                continue;
-                            }
-
                             let btc = request.btc;
                             let xmr = match self.handle_spot_price_request(btc, self.monero_wallet.clone()).await {
-                                Ok(xmr) => xmr,
+                                Ok(xmr) => match xmr {
+                                    Ok(xmr) => xmr,
+                                    Err(e) => {
+                                        tracing::warn!(%peer, "Ignoring spot price request from {} because: {:#}", peer, e);
+                                        match self.swarm.behaviour_mut().spot_price.send_response(channel, spot_price::Response { xmr: None, error: Some(e) }) {
+                                            Ok(_) => {
+                                                continue;
+                                            },
+                                            Err(_) => {
+                                                tracing::debug!(%peer, "Failed to respond with error to spot price request");
+                                                continue;
```

Comment:
Maybe this is a bit nicer: 
```Rust
let result = self.swarm.behaviour_mut().spot_price.send_response(channel, spot_price::Response { xmr: None, error: Some(e) });
if result.is_err() {
     tracing::debug!(%peer, "Failed to respond with error to spot price request");
};
continue;
```

Reply:
![image](https://user-images.githubusercontent.com/5486389/116833390-2a9d6b00-abfc-11eb-8831-05b7d99d8627.png)

Definitely should get rid of this kamehameha!

Reply:
An even better approach would be to extract your own network behaviour for all of this functionality. There is a lot of untested functionality happening here.

Reply:
> An even better approach would be to extract your own network behaviour for all of this functionality. There is a lot of untested functionality happening here.

Done :)
Swarm tests not added yet, will jump on that next.